src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
```

Comment:
We already cool GetAscii twice, as when the first call fails, as it always will, we will get the full length returned and hen cal GetAscii again